sqrl-io/sqrl-io-schema-flexible/src/main/java/com/datasqrl/schema/type/basic/BasicType.java

```diff
@@ -4,10 +4,12 @@
 package com.datasqrl.schema.type.basic;
 
 import com.datasqrl.schema.type.Type;
+import java.util.List;
 
-public interface BasicType<JavaType> extends Type, Comparable<BasicType> {
+public interface BasicType<JavaType> extends Type, Comparable<BasicType<?>> {
 
-  String getName();
+  //First entry is the preferred name, the remaining are for backwards compatibility
+  List<String> getName();
```

Comment:
IMHO a better approach would be to have two methods:
```
//Name of the type
default String getName() { getNames().get(0); }

//List of names for this type (for backwards compatibility)
List<String> getNames()
```

Otherwise you are leaking interface specifics (i.e. that the first name in the list is the default name) to the outside and pushing the responsibility onto the user to enforce it (as you can see by the various calls in AbstractBasicType to getName().get(0))

Also, a method that returns a collection should have a plural name.

Reply:
Good point, got too hung up on the generic issues and forgot about interface

Reply:
I'll actually tie it closer to the planner since it can be pretty expensive to run and the planner can hold the state we would need